submissions/aymkin/OOP-JS (tjsw)/index.js

```diff
@@ -0,0 +1,73 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+
+Code repository: https://github.com/aymkin/a-tiny-JS-world
+Web app: https://aymkin.github.io/a-tiny-JS-world/
+   */
+
+function Inhobitant(species, name, gender, say, friends = []) {
+	this.species = species;
+	this.name = name;
+	this.gender = gender;
+	this.friends = friends;
+	this.say = say;
+}
+
+Inhobitant.prototype.makeSound = function() {
+	return this.say;
+};
+
+Inhobitant.prototype.getFriend = function() {
+	return this.hasOwnProperty('friends') && this.friends.length > 0
+		? `I have friends: ${this.friends.join(', ')}.`
+		: `I have no friends`;
+};
+
+Inhobitant.prototype.getBio = function() {
+	return `I am ${this.species}, my name is <strong>${
+		this.name
+	}</strong>. ${this.getLimbs()}. ${this.getFriend()} \nEvery day I say: ${this.makeSound()}.`;
+};
+
+function Animal(species, name, gender, say, friends) {
+	Inhobitant.call(this, species, name, gender, say, friends);
+	this.legs = 4;
+}
+
+Animal.prototype = Object.create(Inhobitant.prototype);
+
+Animal.prototype.getLimbs = function() {
+	return `I have ${this.legs} legs`;
+};
+
+function Human(species, name, gender, say, friends) {
```

Comment:
`new Human('human',....)` emm... why? We already know it's a human. Why passing `species` at construction time?

You can pass a pre-defined string to a parent class constructor.

Reply:
I got something strange...

Reply:
@OleksiyRudenko did I achieve any of noticed points?

Reply:
I do not resolve any of two comments as far as I am not sure I've done right